vllm/entrypoints/llm.py

```diff
@@ -493,24 +495,51 @@ def _validate_and_add_requests(
                       Sequence[PoolingParams]],
         lora_request: Optional[LoRARequest],
     ) -> None:
+
         if isinstance(inputs, (str, dict)):
             # Convert a single prompt to a list.
             inputs = [inputs]
 
         num_requests = len(inputs)
 
-        if isinstance(params, list) and len(params) != num_requests:
-            raise ValueError("The lengths of prompts and params "
-                             "must be the same.")
+        if params is None:
+            # Use default sampling params.
+            params = [SamplingParams()] * num_requests
+        elif isinstance(params, list):
+            if len(params) != num_requests:
+                raise ValueError("The lengths of prompts and params "
+                                 "must be the same.")
+            if all(isinstance(param, SamplingParams) for param in params):
+                params = [
+                    self._add_guided_processor(param) for param in params
+                    if isinstance(param, SamplingParams)
+                ]
+        elif isinstance(params, SamplingParams):
+            params = self._add_guided_processor(params)
 
-        # Add requests to the engine.
         for i, request_inputs in enumerate(inputs):
             self._add_request(
                 request_inputs,
                 params[i] if isinstance(params, Sequence) else params,
                 lora_request=lora_request,
             )
 
+    def _add_guided_processor(self, params: SamplingParams):
+        if options := params.guided_options:
```

Comment:
**Suggestion**

I would suggest checking for `None` instead. Makes the code more accessible to everyone working on this project.